### PR TITLE
Bug 1939726: Enclose ApplyObject on RetryOnConflict

### PR DESCRIPTION
**Pull request by rcarrillocruz**
This prevents the operator to flip flop on Degraded state
if there is a concurrent update by other external actor.

### CODE REVIEW

**Comment by danwinship** [comment]
/cc @squeed 

In theory we don't do this because CNO will always try to reconcile it again later anyway. Though we should do that without going degraded at least...

I think this change should be fine?

**Comment by rcarrillocruz** [comment]
Yeah, but with the patch I do not see the blip of going Degraded, then reconciling and changing status again, which is what @smarterclayton referred to in the BZ as I understand it.

File: pkg/apply/apply.go [hunk 1]
```diff
@@ -34,32 +35,39 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
...
 
-	if err != nil && apierrors.IsNotFound(err) {
-		log.Printf("does not exist, creating %s", objDesc)
-		err := client.Create(ctx, obj)
+		if err != nil && apierrors.IsNotFound(err) {
+			log.Printf("does not exist, creating %s", objDesc)
+			err := client.Create(ctx, obj)
+			if err != nil {
+				log.Printf("create of %s was unsucessful", objDesc)
+				return err
```
Comment (squeed) [inline pkg/apply/apply.go:46]: Can you still wrap the error here?

Comment (squeed) [reply pkg/apply/apply.go:46]: And in all places.

Comment (squeed) [reply pkg/apply/apply.go:46]: oh crap nevermind. I see why.

File: pkg/apply/apply.go [hunk 2]
```diff
@@ -34,32 +35,39 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
...
+			} else {
+				log.Printf("update was successful")
+			}
+		}
+		return nil
+	})
 
-	return nil
+	return err
```
Comment (squeed) [inline pkg/apply/apply.go:72]: Can you wrap the error here with something a bit more descriptive?

**Comment by squeed** [comment]
seems reasonable. There are probably a few places we update objects directly - it would be worthwhile to grep for `.Update(` and see if there are any other unwrapped ones.

**Pull request merged**